src/ring_theory/polynomial/content.lean

```diff
@@ -0,0 +1,293 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Aaron Anderson
+-/
+import data.finset.gcd
+import data.polynomial
+import data.polynomial.erase_lead
+
+/-!
+# Gauss's Lemma, and GCD structures on polynomials
+
+Gauss's Lemma is one of a few results pertaining to `gcd`s and irreducibility in polynomials over
+GCD domains.
+
+## Main Definitions
+Let `p : polynomial R`.
+ - `p.content` is the `gcd` of the coefficients of `p`.
+ - `p.is_primitive` indicates that `p.content = 1`.
+
+## Main Results
+ - If `p q : polynomial R`, then `(p * q).content = p.content * q.content`.
+
+-/
+
+variables {R : Type*} [integral_domain R]
+
+namespace polynomial
+section gcd_monoid
+variable [gcd_monoid R]
```

Comment:
How hard will it be to get statements about irreducibility in the case of a UFD? Will this `gcd_monoid` tie your hands in the future?

Reply:
As far as getting my hands tied, it's possible that `is_primitive` could be defined in a more general context, but it should be provably equal. I don't think `content` will be defined in any more general context, although perhaps a `content_ideal` could be.

As far as UFD results, I think it'll be pretty easy to provide a `gcd_monoid` instance on `polynomial R`, but I haven't tried that yet.

Reply:
The next thing to do will actually be to fix how `normalization_monoid`s work on polynomials